### PR TITLE
 Implement query execution and generation for CrateDB. 

### PR DESCRIPTION
**Pull request by kovrus**
## Summary of the changes / Why this is an improvement

The PR contains query generation and execution logic for CrateDB.

## Checklist

 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: cmd/tsbs_generate_queries/databases/cratedb/devops.go [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+		LIMIT 5`,
+		interval.EndUnixMillis())
+
+	humanLabel := "CrateDB max cpu over last 5 min-intervals (random end)"
+	humanDesc := fmt.Sprintf("%s: %s", humanLabel, interval.EndString())
+	d.fillInQuery(qi, humanLabel, humanDesc, sql)
+}
+
+// LastPointPerHost finds the last row for every host in the dataset
+func (d *Devops) LastPointPerHost(qi query.Query) {
```
Comment (kovrus) [inline cmd/tsbs_generate_queries/databases/cratedb/devops.go:131]: not sure about this query

File: query/cratedb_test.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+	// Since we use a pool, check that the next one is reset
+	tq = NewCrateDB()
+	check(tq)
+	tq.Release()
+}
+
+func TestCrateDBSetAndGetID(t *testing.T) {
+	for i := 0; i < 2; i++ {
+		q := NewTimescaleDB()
```
Comment (andreidan) [inline query/cratedb_test.go]: NewCrateDB

**Review by andreidan — APPROVED** [review]
lgtm - left one comment

File: cmd/tsbs_generate_queries/databases/cratedb/devops.go (hunk 1 above)
Comment (mfussenegger) [reply cmd/tsbs_generate_queries/databases/cratedb/devops.go:131]: This doesn't seem right to me. `DISTINCT` is global over all columns. 

We'd have to have some kind of GROUP BY?

Comment (mfussenegger) [reply cmd/tsbs_generate_queries/databases/cratedb/devops.go:131]: https://github.com/crate/tsbs/blob/19932e723e844ad33cff0ed73c4a994568dc75bc/cmd/tsbs_generate_queries/databases/influx/devops.go#L128

**Review by mfussenegger — APPROVED** [review]
otherwise lgtm

File: cmd/tsbs_generate_queries/databases/cratedb/devops.go (hunk 1 above)
Comment (kovrus) [reply cmd/tsbs_generate_queries/databases/cratedb/devops.go:131]: it seems that it has to be smth like that 

```
SELECT * 
FROM (select tags['hostname'], max(ts) as ts from cpu group by tags['hostname']) t, cpu 
WHERE t.ts = cpu. ts AND t.tags['hostname'] = cpu.tags['hostname'];
```

**Pull request merged**